### PR TITLE
Add rotate certificate input type

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

File: apis/management.cattle.io/v3/cluster_types.go [hunk 1]
```diff
@@ -302,3 +303,8 @@ type IngressCapabilities struct {
 	IngressProvider      string `json:"ingressProvider,omitempty"`
 	CustomDefaultBackend bool   `json:"customDefaultBackend,omitempty"`
 }
+
+type RotateCertificateInput struct {
+	RotateCACertificates bool     `json:"rotateCACertificates,omitempty" norman:"default=false"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_types.go]: I don't think we need Rotate* prepending the fields; should be just CACertificates, and Components
default=false is not needed. All bools are false by default

File: apis/management.cattle.io/v3/cluster_types.go [hunk 2]
```diff
@@ -302,3 +303,8 @@ type IngressCapabilities struct {
...
+	RotateComponents     []string `json:"rotateComponents,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_types.go]: Would empty list mean all components/services?

File: apis/management.cattle.io/v3/cluster_types.go (hunk 1 above)
Comment (galal-hussein) [reply apis/management.cattle.io/v3/cluster_types.go]: sure i will do this change

File: apis/management.cattle.io/v3/cluster_types.go (hunk 2 above)
Comment (galal-hussein) [reply apis/management.cattle.io/v3/cluster_types.go]: yeah that's what it means

Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_types.go]: we need to list available values - etcd, control, etc - using `norman:"type=array[enum],options=....` annotation

File: apis/management.cattle.io/v3/cluster_types.go [hunk 3]
```diff
@@ -74,6 +74,7 @@ type ClusterSpec struct {
 	DefaultClusterRoleForProjectMembers  string                               `json:"defaultClusterRoleForProjectMembers,omitempty" norman:"type=reference[roleTemplate]"`
 	DockerRootDir                        string                               `json:"dockerRootDir,omitempty" norman:"default=/var/lib/docker"`
 	EnableNetworkPolicy                  *bool                                `json:"enableNetworkPolicy" norman:"default=false"`
+	RotateCertificates                   *RotateCertificateInput              `json:"rotateCertificates,omitempty"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/cluster_types.go]: should rotateCertificates be a part of rkeConfig? as it's RKE clusters feature only

**Pull request merged**